orangecontrib/text/preprocess/normalize.py

```diff
@@ -0,0 +1,81 @@
+from nltk import stem
+
+__all__ = ['BaseNormalizer', 'WordNetLemmatizer', 'PorterStemmer',
+           'SnowballStemmer', 'DictionaryLookupNormalizer']
+
+
+class BaseNormalizer:
+    name = NotImplemented
+    normalizer = NotImplemented
+
+    def __call__(self, tokens):
+        """ Normalizes tokens to canonical form. """
+        if isinstance(tokens, str):
+            return self.normalize(tokens)
+        return [self.normalize(token) for token in tokens]
+
+    def normalize(self, token):
+        return self.normalizer(token)
+
+    def configuration_changed(self):
+        pass
+
+    def __str__(self):
+        return self.name
+
+
+class WordNetLemmatizer(BaseNormalizer):
+    name = 'WordNet Limmatizer'
+    normalizer = stem.WordNetLemmatizer().lemmatize
+
+
+class DictionaryLookupNormalizer(BaseNormalizer):
+    """ Normalizes token with a <token: canonical_form> dictionary. """
+    name = 'Dictionary Lookup'
+
+    def __init__(self, dictionary):
+        super().__init__()
+        self._dictionary = dictionary
+
+    @property
+    def dictionary(self):
+        return self._dictionary
+
+    @dictionary.setter
+    def dictionary(self, value):
+        self._dictionary = value
+        self.configuration_changed()
+
+    def normalize(self, token):
+        return self.dictionary.get(token, token)
+
+
+class PorterStemmer(BaseNormalizer):
+    name = 'Porter stemmer'
```

Comment:
If WordNet Lemmatizer is in camel case then stemmer shoud be also. Same goes for Snowball stemmer.